### PR TITLE
[WiP][CR]Mi-go with a gun

### PR DESCRIPTION
**Pull request by Coolthulhu**
Mi-gos are supposed to be intelligent. And intelligent beings don't kill things with claws.

This PR adds a mi-go variant which uses a "death ray" - small rifle with amazing accuracy, huge armor piercing, good damage, but no damage scaling on good hits.
When (if) you manage to kill the alien, you get to use its gun for 10 shots. After that it's spent and can't be reloaded.

Unlike robots, which warn you and then kill you, mi-go gunner will warn you by firing at you (almost invariably hitting, piercing all armor, and dealing ~50 damage), then waiting for half a minute to let you run.

Currently spawns only in labs where you can close the door on them.
Shines in the dark to let you know what is shooting at you, but has enough night vision not to let you shoot back without risking a hit.

### CODE REVIEW

**Comment by BorkBorkGoesTheCode** [comment]
I thought they were parrots.

**Comment by Noctifer-de-Mortem** [comment]
Limited number of "patrol" / "scout" mi-gos (Is the plural with an s or just mi-go?) cloud spawn  outside the labs along with the normal mi-gos using pistol variants of this weapon.

**Comment by Coolthulhu** [comment]
Now that I think about it, it needs some sort of grace period to avoid instantly taking out players who just enter the room.

**Comment by Noctifer-de-Mortem** [comment]
It could just not shoot the turn it notices the player but rather the turn after.

**Comment by Mecares** [comment]
How about letting it repeat warnings like: "Careful this one has some kind of gun" "What is that in its hand ARGH"

File: data/json/items/ammo/alien.json [hunk 1]
```diff
@@ -0,0 +1,15 @@
+[
+  {
+    "id": "death_bolt",
```
Comment (Rivet-the-Zombie) [inline data/json/items/ammo/alien.json]: This should probably have the NEVER_MISFIRES flag as well.

**Comment by Voskhod-1** [comment]
Any particular reason why the mi-gunner waits for half a minute to let the player run? I figure they're shooting to kill, or are they shooting to scare the player off?

**Comment by pisskop** [comment]
this highlights the current issue with guns.

theyre deadly.


and beside, a door opening, 100morale 100aggro migo is not something a player can outrun without pain, nevermind with it

**Comment by jokermatt999** [comment]
I really like the idea about warning shouts. It definitely would be creepier and more fair to the players. Nether creatures and zombies are opposing factions, so in a lab setting they'd probably spend at least some of the time gunning down zombies. Labs really ought to have some even more challenging enemies in them.

**Comment by DangerNoodle** [comment]
I am unsure about this idea, but I do like the idea of fleshing out nether monsters, especially intelligent ones like the mi-go.

Alternatively, from a lore standpoint it would help if we had someone who has read the H.P. Lovecraft works the creatures appeared in, for flavor ideas regarding what other signs of intelligence they could display. I do know that their fungal biology is at least referenced in the source code, by way of immunity to fungal conversion.

**Comment by Mecares** [comment]
> Alternatively, from a lore standpoint it would help if we had someone who has read the H.P. Lovecraft works 
Well there is http://lovecraft.wikia.com/wiki/Mi-go

**Comment by Rivet-the-Zombie** [comment]
>this highlights the current issue with guns.
>
>theyre deadly.

Then they're working as intended.

**Comment by Coolthulhu** [comment]
> Any particular reason why the mi-gunner waits for half a minute to let the player run? I figure they're shooting to kill, or are they shooting to scare the player off?

They're supposed to just scare the player away from some mysterious thing they're working on.

I'd like it to change from warning shots to full auto if player fights back, but there is no way to do that yet.

> and beside, a door opening, 100morale 100aggro migo is not something a player can outrun without pain, nevermind with it

It's 80 speed, mostly so that it can be outran with pain.

> Alternatively, from a lore standpoint it would help if we had someone who has read the H.P. Lovecraft works the creatures appeared in, for flavor ideas regarding what other signs of intelligence they could display.

They're supposed to be slavers with 3 castes: workers, scientists, soldiers.
They're skilled in surgery.
I don't think they have any weapons in the books themselves, but in derived works they commonly get a lightning gun.
The weapon here is supposed to be more of an adapted tool than a proper weapon. Something that helps them do all the work they're getting on here, because it would be pretty dumb to do everything by hand.

**Comment by Rivet-the-Zombie** [comment]
>some mysterious thing they're working on

Stolen brains?

**Comment by Noctifer-de-Mortem** [comment]
> I'd like it to change from warning shots to full auto if player fights back, but there is no way to do that yet.

Why not give them low aggression and a clam trigger when the player runs away. so they just shoot once or twice but then make them give them the anger trigger for when they are hurt or their friends are hurt.

**Comment by Coolthulhu** [comment]
> Why not give them low aggression and a clam trigger when the player runs away.

There is no good measure of player running away.

**Comment by pisskop** [comment]
>.>
<.<

We need migo npcs

**Comment by Noctifer-de-Mortem** [comment]
> There is no good measure of player running away.

Just check and you are right, I was confusing it with placate_trigger  `"PLAYER_WEAK"`.

And that placate would not suffice... The best to do would be to give it low aggression or/and moral.

**Comment by DangerNoodle** [comment]
If the weapon has lower damage and more effective at stunning, like having the beanbag effect, that might increase its ability to deter the player with less tendency of instant death. It would still be dangerous if the player is still being hounded by other mi-gos though.

**Comment by pisskop** [comment]
Stunning isnt a bad option.  Its very nature is a deterrent.  Just make sure its got more competence than that pack of AHorrors that pose as a boss.

**Comment by jokermatt999** [comment]
Stunning would also make sense if they have lightning rifles in other media. Reuse the shocker attack would be one option I guess.

This whole thread makes me think I ought to spend Halloween week re-reading some Lovecraft.

**Comment by BorkBorkGoesTheCode** [comment]
Reducing damage and adding BEANBAG seems like a better idea.

**Comment by DangerNoodle** [comment]
> This whole thread makes me think I ought to spend Halloween week re-reading some Lovecraft.

A shoggoth would be an amusing costume idea.

**Comment by Coolthulhu** [comment]
50 damage isn't instakill. It can't go above 50.
In fact, stun would most likely have a higher insta-kill potential.

> The best to do would be to give it low aggression or/and moral.

Low aggression and morale would just mean it would wait for the player to approach and gun it down.

**Comment by pisskop** [comment]
The idea that the migo doesnt want to chase you down and murderface you doesnt stop it from prefering you dead.

Phasers set to kill is fine.  Wed need more ai tags to properly emulate a defensive mobile monster.

80 speed plus all the closeable doors in labs makes this doable.  more doable than making, say, a monster with an arm cannon that does the same thing.

**Comment by Coolthulhu** [comment]
Pretty sure that a neutral mi-go wouldn't even target the player.
Unless it would, then it could be a good idea.

**Comment by DangerNoodle** [comment]
If it was set to "player close" as an attack trigger, as well as being hurt, and possibly noise. A noise trigger would have odd results if it was an anger trigger, but would ensure it reacts to being fired upon if the player missed.

**Comment by jokermatt999** [comment]
What about noise triggering the stalking behavior animals use when the player is low health? Never outright attacking unless cornered or provoked, but keeping an eye on the player? Give them some gradual decay on their morale in that state, and they should "flee" back to doing whatever terrible mi-go experiments they get up to.

**Comment by Voskhod-1** [comment]
Plugging a suggestion for mi-go's and other Lovecraft tier monsters

#18957

**Comment by Nacarbac** [comment]
One other common weapon that derived works (in particular the Delta Green and Call of Cthulhu tabletop games) associate with Mi-Go is a kind of "cryogenic mist launcher" capable of freezing a human solid in short order - which could be an interesting spin on the toxic fog effects already present in the game.

**Comment by pisskop** [comment]
Customizable gas?  -> yes please

**Comment by DangerNoodle** [comment]
It would be nice if we had freezing weapons as a ranged effect, but coding it might be awkward. Unless there was a field added that lowered temperature?

**Comment by Coolthulhu** [comment]
I won't have the time to work on this.

Anything worth salvaging from here?
The alien gun perhaps? Could be dropped in some lab location, for example randomly placed outside the "box" in the finale for easy picking.

**Comment by Rivet-the-Zombie** [comment]
I like the entire thing as it currently is.

**Comment by Coolthulhu** [comment]
I don't like how it can blast someone going around a corner.
If it only shot those who can see it, it could work.

**Comment by Rivet-the-Zombie** [comment]
>I don't like how it can blast someone going around a corner.

I just got done playing S.T.A.L.K.E.R. for the last 45 minutes; this doesn't bother me, though I can see how it would bother other people.

>If it only shot those who can see it, it could work.

I'm rather surprised monsters don't have some sort of `aim_cost` stat already.

**Comment by Rivet-the-Zombie** [comment]
>I don't like how it can blast someone going around a corner.

I just got done playing S.T.A.L.K.E.R. for the last 45 minutes; this doesn't bother me. Though I can see how it would bother other people.

>If it only shot those who can see it, it could work.

I'm rather surprised monsters don't have some sort of `aim_cost` stat already.

**Comment by DangerNoodle** [comment]
Or if the "you are being targeted" message was not restricted to antimaterial turrets and tank bots. Or is that available in JSON format now?

**Comment by pisskop** [comment]
id rather not have more button pressing in dire situation.

50 damage on a bp is equivalent to amy other gun

**Comment by mugling** [comment]
> I'm rather surprised monsters don't have some sort of aim_cost stat already.

This is a good suggestion and warrants further investigation. In particular robots could change to use the same aiming rules as the player which could help balance them and avoid that pseudo-targeting stage they currently have?

**Comment by BorkBorkGoesTheCode** [comment]
After thinking, the only problem I see is the 1000 pierce rating. If it was removed, the player could have a chance to survive.

**Comment by Coolthulhu** [comment]
No, I'd rather lower damage.
Pierce rating only means it skips armor. That's supposed to be a major part of the gimmick.

**Comment by DangerNoodle** [comment]
Lower damage with the same pierce likely would be sensible, yes.

Additionally, the warning message that it could generate beforehand does not have to full-on interrupt the player like with tank drones. Though it would ensure the player gets the hint, a basic warning like the beeping of standard turrets would cause less player annoyance (unless failing to notice it needs to YASD).

**Comment by Coolthulhu** [comment]
I'm closing this for now.
I'll probably revisit the idea when we have variable gun damage types, like heat for lasers. Then death ray would be biological and would skip armors and robots, which would be a cool gimmick for it.

**Comment by pisskop** [comment]
biological damage should be restricted to those that have biological parts.

and thus robots, zombies, and to an extent fungus would be immune somewhat.

**Comment by DangerNoodle** [comment]
It does on what you mean by biological. A heat ray sounds more like something that would be mitigated by insulative or reflective gear (entry suit?), but if it remains highly armor-piercing that suggests something able to bypass isulation via superscience.

Biological sounds more like either the weapon is organic technologically (which sounds like the intended interpretation, given heat rays were mentioned in the same paragraph), or a biological weapon which does indeed make sense to only affect organic enemies.

**Comment by Rivet-the-Zombie** [comment]
>biological damage should be restricted to those that have biological parts.
>
>and thus robots, zombies, and to an extent fungus would be immune somewhat.

Zombies produce body heat and are *made of* biological parts.

**Comment by DangerNoodle** [comment]
I can see the idea of a biological weapon, using the correct definition, being ineffective against zombies. They do seem to decay at a reduced rate, which is a caveat that is part and parcel of zombie movies and games.

**Comment by pisskop** [comment]
Zombies are made of blob.  They have limited use for organs.

bio damage is described as not damage to biologics but damage to internal tissue.

**Pull request closed without merging**